### PR TITLE
[Hub Generated] Review request for Microsoft.Security to add version preview/2023-07-01-preview

### PR DESCRIPTION
**Pull request by v-tamarstern**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/v-tamarstern-security-Microsoft.Security-2023-07-01-preview...main/).
### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version). Note that this doesn't apply if you are trying to merge a PR that was previously in the private repository.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/overview_breakingchanges).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](http://aka.ms/aprwiki))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applicationMappingRules.json [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+          "type": "array",
+          "description": "The mapping tag key list",
+          "items": {
+            "description": "The mapping tag key",
+            "type": "string"
+          }
+        },
+        "includedMappingTagValues": {
+          "type": "array",
+          "description": "The Included mapping tag values,For an empty array - all the values should create applications",
```
Comment (SapirElmakayes) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applicationMappingRules.json:295]: lower case

Comment (SapirElmakayes) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applicationMappingRules.json:295]: lower case

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applicationMappingRules.json [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+          "description": "Excluded Scopes, filter out the descendants of the scope",
+          "items": {
+            "description": "The excluded scope",
+            "type": "string"
+          },
+          "x-ms-identifiers": []
+        },
+        "includeMemberScopes": {
+          "type": "boolean",
+          "description": "Defines whether the rule is management scope rule (master connector as a single scope or management scope)"
```
Comment (SapirElmakayes) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applicationMappingRules.json:312]: Management connector

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applicationMappingRules.json [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+      }
+    }
+  },
+  "parameters": {
+    "RuleId": {
+      "name": "ruleId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The rule Key - Unique key for the rule for ABC (GUID)",
```
Comment (SapirElmakayes) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applicationMappingRules.json:352]: lower case

Comment (SapirElmakayes) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applicationMappingRules.json:352]: no need the "for ABC"

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+          "type": "string",
+          "description": "The application mapping rule Key, the key of the rule were created the application (GUID)"
+        },
+        "kqlQuery": {
+          "type": "string",
+          "description": "The application mapping KQL query"
+        },
+        "applicationOwner": {
+          "type": "string",
+          "description": "application owner (GUID)"
```
Comment (SapirElmakayes) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json:272]: Upper case

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+      }
+    }
+  },
+  "parameters": {
+    "RuleId": {
+      "name": "ruleId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The rule Key - Unique key for the rule for ABC (GUID)",
```
Comment (SapirElmakayes) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json:300]: no need

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+      "x-ms-parameter-location": "method"
+    },
+    "ApplicationId": {
+      "name": "applicationId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The rule Key - Unique key for the rule for ABC (GUID)",
```
Comment (SapirElmakayes) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json:308]: Change to application

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+        "$ref": "#/definitions/Application"
+      },
+      "x-ms-parameter-location": "method"
+    },
+    "Scope": {
+      "name": "scope",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The scope of the application mapping rules rules. Valid scopes are: management group (format: 'providers/Microsoft.Management/managementGroups/{resourceName}'), subscription (format: 'subscriptions/{subscriptionId}'), or security connector (format: 'subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Security/securityConnectors/{resourceName})'",
```
Comment (SapirElmakayes) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json:326]: applicaiton

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/examples/ApplicationMappingRules/GetSecurityConnectorApplicationMappingRule_example.json [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/resourceGroups/gcpResourceGroup/providers/Microsoft.Security/applicationMappingRules/ad9a8e26-29d9-4829-bb30-e597a58cdbb8",
+        "name": "ad9a8e26-29d9-4829-bb30-e597a58cdbb8",
+        "type": "Microsoft.Security/applicationMappingRules",
+        "properties": {
+          "tenantId": "f0b6d37b-e4bc-4719-9291-c066c3194f23",
+          "environmentId": "subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/resourceGroups/gcpResourceGroup",
```
Comment (SapirElmakayes) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/examples/ApplicationMappingRules/GetSecurityConnectorApplicationMappingRule_example.json:15]: I'm not sure if the resource group is a valid environment, could you please validate it with Roi / Governance stream?

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/examples/HealthReports/GetHealthReports_example.json [hunk 9]
```diff
@@ -0,0 +1,56 @@
+{
+  "parameters": {
```
Comment (SapirElmakayes) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/examples/HealthReports/GetHealthReports_example.json:2]: revert? :)

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/examples/HealthReports/ListHealthReports_example.json [hunk 10]
```diff
@@ -0,0 +1,47 @@
+{
+  "parameters": {
+    "api-version": "2023-07-01-preview",
```
Comment (SapirElmakayes) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/examples/HealthReports/ListHealthReports_example.json:3]: revert?

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/healthReports.json [hunk 11]
```diff
@@ -0,0 +1,373 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Microsoft Defender for Cloud",
```
Comment (SapirElmakayes) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/healthReports.json:4]: revert?

**Comment by v-tamarstern** [comment]
Hi @Wzb123456789 
Appreciate your review
The Swagger Avocado Failing is unaffected by my additions
Thanks

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json [hunk 12]
```diff
@@ -0,0 +1,326 @@
...
+        },
+        "properties": {
+          "description": "Properties of an application business criticality",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
```
Comment (ms-henglu) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json:242]: Consider use `../../../../../common-types/resource-management/v5/types.json#/definitions/Resource`, it alreay has `systemData` defined inside.

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json [hunk 13]
```diff
@@ -0,0 +1,326 @@
...
+    },
+    "ApplicationProperties": {
+      "type": "object",
+      "description": "Describes properties of the application",
+      "properties": {
+        "tenantId": {
+          "type": "string",
+          "description": "The tenant Id associated with the application (GUID)"
+        },
+        "environmentId": {
```
Comment (ms-henglu) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json:254]: From the example, this field is alawys equal to the scope, is it duplicated with the scope? Or is there other use case?

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json [hunk 14]
```diff
@@ -0,0 +1,326 @@
...
+          "type": "string",
+          "description": "The application environment Id"
+        },
+        "applicationMappingRuleId": {
```
Comment (ms-henglu) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json:258]: This field can be retrieved from the API path, is it necessary to define it here?

**Review by ms-henglu — CHANGES_REQUESTED** [review]

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/examples/ApplicationMappingRules/ListBySecurityConnectorApplicationMappingRules_example.json [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+    "200": {
+      "body": {
+        "value": [
+          {
+            "id": "subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/resourceGroups/gcpResourceGroup/providers/Microsoft.Security/applicationMappingRules/ad9a8e26-29d9-4829-bb30-e597a58cdbb8",
+            "name": "ad9a8e26-29d9-4829-bb30-e597a58cdbb8",
+            "type": "Microsoft.Security/applicationMappingRules",
+            "properties": {
+              "tenantId": "f0b6d37b-e4bc-4719-9291-c066c3194f23",
+              "environmentId": "subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/resourceGroups/gcpResourceGroup",
```
[the remark this answers is not in the document]
Comment (v-tamarstern) [reply specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/examples/ApplicationMappingRules/ListBySecurityConnectorApplicationMappingRules_example.json:16]: fix it

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json [hunk 16]
```diff
@@ -0,0 +1,326 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/{scope}/providers/Microsoft.Security/applicationMappingRule/{ruleId}/applications/{applicationId}": {
```
[the remark this answers is not in the document]
Comment (v-tamarstern) [reply specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json:37]: In this case, the API allows customers to only update resources using a "PUT" call (modifying or making changes to the existing information associated with that resource - Business criticality/owner). However, the API does not provide options for creating or deleting resources directly. The creation and deletion of resources are handled by a background service, which is a separate component or process that operates behind the scenes. What is the correct way to define it in the swagger?

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json [hunk 17]
```diff
@@ -0,0 +1,312 @@
...
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.Security/applications": {
+      "get": {
+        "x-ms-examples": {
+          "Get a list of applications": {
```
Comment (ms-henglu) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json:125]: The description is not accurate.

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json [hunk 18]
```diff
@@ -0,0 +1,312 @@
...
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.Security/applicationMappingRule/{ruleId}/applications": {
+      "get": {
+        "x-ms-examples": {
+          "Get list of  applications": {
```
Comment (ms-henglu) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json:164]: The description is not accurate.

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json [hunk 19]
```diff
@@ -0,0 +1,312 @@
...
+          "Application"
+        ],
+        "description": "Get a list of all relevant applications over a rule Id.",
+        "operationId": "Application_ListByRuleId",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/RuleId"
```
Comment (ms-henglu) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json:178]: The parameters:ruleId,scope should be kept in the same order as they present in the path.

refs: https://github.com/Azure/azure-openapi-validator/blob/main/docs/parameter-order.md

File: specification/security/resource-manager/readme.md [hunk 20]
```diff
@@ -905,7 +924,7 @@ override-info:
 
 These settings apply only when `--tag=package-2023-05` is specified on the command line.
 
-```yaml $(tag) == 'package-2023-05'
+``` yaml $(tag) == 'package-2023-05'
 input-file:
   - Microsoft.Security/stable/2023-05-01/serverVulnerabilityAssessmentsSettings.json
```
Comment (ms-henglu) [inline specification/security/resource-manager/readme.md:930]: Missing ``` to close the code block here.

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applicationMappingRules.json [hunk 21]
```diff
@@ -0,0 +1,374 @@
...
+        },
+        "properties": {
+          "description": "Properties of an application mapping rule",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationMappingRuleProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
```
[the remark this answers is not in the document]
Comment (mentat9) [reply specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applicationMappingRules.json:255]: Need newer than v3 to get systemData, should use v5 here.

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applicationMappingRules.json [hunk 22]
```diff
@@ -0,0 +1,374 @@
...
+        "excludedScopes": {
+          "type": "array",
+          "description": "Excluded Scopes, filter out the descendants of the scope",
+          "items": {
+            "description": "The excluded scope",
+            "type": "string"
+          },
+          "x-ms-identifiers": []
+        },
+        "includeMemberScopes": {
```
[the remark this answers is not in the document]
Comment (mentat9) [reply specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applicationMappingRules.json:304]: +1

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json [hunk 23]
```diff
@@ -0,0 +1,318 @@
...
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Updated",
+            "schema": {
+              "$ref": "#/definitions/Application"
+            }
+          },
+          "201": {
```
Comment (mentat9) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json:113]: >201 [](http://example.com/codeflow?start=11&length=3)

Did you add the create scenario?

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json [hunk 24]
```diff
@@ -0,0 +1,318 @@
...
+      "properties": {
+        "properties": {
+          "description": "Properties of an application business criticality",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
```
Comment (mentat9) [inline specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json:242]: >v1 [](http://example.com/codeflow?start=35&length=2)

Need v3 or later to get systemData, should use latest (v5).

File: specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json (hunk 16 above)
[the remark this answers is not in the document]
Comment (mentat9) [reply specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json:37]: Couple of questions:
1. Why does your service not support create and delete via API? These are basic operations required by ARM resource types.
2. What scopes are supported by this API?

**Comment by mentat9** [comment]
@v-tamarstern - Please fill out the submission form above: it's required for ARM review and proper handling of your PR.

**Comment by eliagrady** [comment]
please remove any unrelated file changes from this PR (spacing, formatting etc.)

**Pull request closed without merging**